tests/qos/test_qos_sai.py

```diff
@@ -1915,11 +1929,20 @@ def testQosSaiQWatermarkAllPorts(
 
         allTestPorts = []
         allTestPortIps = []
-        all_dst_info = dutConfig['testPortIps'][get_src_dst_asic_and_duts['dst_dut_index']]
-        allTestPorts.extend(list(all_dst_info[get_src_dst_asic_and_duts['dst_asic_index']].keys()))
-        allTestPortIps.extend([
-            x['peer_addr'] for x in
-            all_dst_info[get_src_dst_asic_and_duts['dst_asic_index']].values()])
+        testPortIps = dutConfig["testPortIps"]
+        if separated_dscp_to_tc_map_on_uplink:
+            # Remove the upstream ports from the test port list.
+            allTestPorts = list(set(allTestPorts) - set(dutConfig['testPorts']['uplink_port_ids']))
```

Comment:
in this condition branch, "allTestPorts" is not initialized. need to modify, and can reference below changs:

```
         allTestPorts = []
         allTestPortIps = []
         testPortIps = dutConfig["testPortIps"]
-        if separated_dscp_to_tc_map_on_uplink:
+        all_dst_info = dutConfig['testPortIps'][get_src_dst_asic_and_duts['dst_dut_index']]
+        allTestPorts.extend(list(all_dst_info[get_src_dst_asic_and_duts['dst_asic_index']].keys()))
+        allTestPortIps.extend([
+            x['peer_addr'] for x in
+            all_dst_info[get_src_dst_asic_and_duts['dst_asic_index']].values()])
+        if separated_dscp_to_tc_map_on_uplink(dut_qos_maps):
             # Remove the upstream ports from the test port list.
             allTestPorts = list(set(allTestPorts) - set(dutConfig['testPorts']['uplink_port_ids']))
             allTestPortIps = [
                 testPortIps[get_src_dst_asic_and_duts['dst_dut_index']]
                 [get_src_dst_asic_and_duts['dst_asic_index']][port]['peer_addr']
                 for port in allTestPorts]
-        else:
-            all_dst_info = dutConfig['testPortIps'][get_src_dst_asic_and_duts['dst_dut_index']]
-            allTestPorts.extend(list(all_dst_info[get_src_dst_asic_and_duts['dst_asic_index']].keys()))
-            allTestPortIps.extend([
-                x['peer_addr'] for x in
-                all_dst_info[get_src_dst_asic_and_duts['dst_asic_index']].values()])
         try:
             tc_to_q_map = dut_qos_maps['tc_to_queue_map']['AZURE']
             tc_to_dscp_map = {v: k for k, v in dut_qos_maps['dscp_to_tc_map']['AZURE'].items()}
@@ -1952,8 +1955,8 @@ class TestQosSai(QosSaiBase):
                 "and key AZURE to run this test.")
         dscp_to_q_map = {tc_to_dscp_map[tc]: tc_to_q_map[tc] for tc in tc_to_dscp_map}
         if get_src_dst_asic_and_duts['single_asic_test']:
-            allTestPorts.remove(dutConfig["testPorts"]["src_port_id"])
-            allTestPortIps.remove(dutConfig["testPorts"]["src_port_ip"])
+            if dutConfig["testPorts"]["src_port_id"] in allTestPorts: allTestPorts.remove(dutConfig["testPorts"]["src_port_id"])
+            if dutConfig["testPorts"]["src_port_ip"] in allTestPortIps: allTestPortIps.remove(dutConfig["testPorts"]["src_port_ip"])

         testParams = dict()
         testParams.update(dutTestParams["basicParams"])
```